File: back/app/models/dish.rb
class Dish
    include Mongoid::Document
    field :name, type: String
    field :description, type: String
    field :components, type: String
    field :score, type: Float
    field :calories, type: Integer
    field :proteins, type: Integer
    field :fats, type: Integer
    field :carbohydrates, type: Integer
    field :price, type: Integer
    field :sale_price, type: Integer
    field :categories, type: Array, default: []
    field :cuisines, type: Array, default: []
    field :restrictions, type: Array, default: []

    validates :name, length: {minimum: 1, maximum: 64}, presence: true
    validates :description, length: {minimum: 1, maximum: 2048}, presence: true
    validates :description, length: {minimum: 1, maximum: 2048}, presence: true
    validates :components, length: {minimum: 1, maximum: 2048}, presence: true    
    validates :calories, inclusion: 0..100000
    validates :proteins, inclusion: 0..100000
    validates :fats, inclusion: 0..100000
    validates :carbohydrates, inclusion: 0..100000
    validates :price, inclusion: 0..1000000

    belongs_to :image
    belongs_to :place

  end
  

File: mobile/lib/models/dish.dart
class Dish {
  
  final String id;
  final String name;
  final String description;
  final String imageId;
  final String restaurantId;
  final String currency;
  final String components;

  final double price;
  final double salePrice;
  final int calories;
  final int proteins;
  final int fats;
  final int carbohydrates;

  final List<String> categories;

  Dish(
    {
      this.id, 
      this.name, 
      this.description, 
      this.price, 
      this.salePrice,
      this.imageId,
      this.calories, 
      this.restaurantId, 
      this.currency,
      this.fats,
      this.carbohydrates,
      this.proteins,
      this.components,
      this.categories = const []
    }
  );

  factory Dish.fromJson(Map<String, dynamic> json) {
    return Dish(
      id: json['_id'],
      name: json['name'],
      description: json['description'],
      imageId: json['image_id'],
      price: json['price'] / 100.0,
      salePrice: json['sale_price'] != null ? json['sale_price'] / 100 : json['price'] / 100,
      calories: json['calories'],
      proteins: json['proteins'],
      fats: json['fats'],
      carbohydrates: json['carbohydrates'],
      restaurantId: json['restaurant_id'],
      currency: json['currency'],
      components: json['components'],
      categories:  List<String>.from(json['categories']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'name': name,
      'description': description,
      'image_id' : imageId,
      'price': price,
      'calories': calories,
      'restaurant_id': restaurantId,
      'currency': currency,
      'categories': categories
    };
  }

  @override 
  int get hashCode => id.hashCode;

  bool operator ==(o) => o is Dish && o.id == id;

}